### DISCUSSION TITLE
query format

### DISCUSSION THREAD

**Discussion by max397574** [nvim-treesitter/nvim-treesitter#4105] [General]
is there a general way of how you format the queries (I'm working on an automatic formatter atm)

like
when a newline, how many spaced indent etc.
to me this seems kinda random rn

**Comment by theHamsta** [comment]
In general we try to follow Lisp indenting conventions. I guess we should use Emacs. So a formatter for Scheme or Common Lisp or would be fine. `[ ]` are often formatted a bit differently

**Comment by max397574** [comment]
to me things seem quite inconsistent
https://github.com/nvim-treesitter/nvim-treesitter/blob/68e8181dbcf29330716d380e5669f2cd838eadb5/queries/lua/highlights.scm#L134-L135
3 spaces

https://github.com/nvim-treesitter/nvim-treesitter/blob/68e8181dbcf29330716d380e5669f2cd838eadb5/queries/rust/highlights.scm#L49-L50
2 spaces

perhaps it's just because I'm not familiar with lisp indents
but this seems to me like it should have the same indents
